### ISSUE TITLE
[iOS] `ITSAppUsesNonExemptEncryption` in Info.plist keeps getting removed during pod install

### ISSUE THREAD

**Issue by annuh** [facebook/react-native#41898]
### Description

When running `pod install` with RN 0.73, the existing content under the key`ITSAppUsesNonExemptEncryption` in Info.plist will always be completely removed:

```diff
-	<key>ITSAppUsesNonExemptEncryption</key>
-	<string>NO</string>
```

I'm using this to bypass the “Missing export compliance information” warning in the App Store (https://joeybronner.medium.com/bypass-the-missing-export-compliance-information-warning-on-app-store-b1e1c54606b6).

It looks like this issue a quite similar to https://github.com/facebook/react-native/issues/41874.

### Steps to reproduce

- Init RN (ie `npx react-native@latest init AwesomeProject`)
- Install pods (ie ` pod install --project-directory=ios`)
- Add `ITSAppUsesNonExemptEncryption` key to `./ios/AwesomeProject/Info.plist`)
- Re-install pods

Now the `ITSAppUsesNonExemptEncryption` key is removed from the Info.plist file.

### React Native Version

0.73.0

### Affected Platforms

Build - MacOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 114.75 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.9.0
    path: ~/.nvm/versions/node/v20.9.0/bin/node
  Yarn:
    version: 1.22.19
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.1.0
    path: ~/.nvm/versions/node/v20.9.0/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.14.3
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.0
      - iOS 17.0
      - macOS 14.0
      - tvOS 17.0
      - watchOS 10.0
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10406996
  Xcode:
    version: 15.0.1/15A507
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.9
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.0
    wanted: 0.73.0
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```
```


### Stacktrace or Logs

```text
None
```

### Reproducer

https://github.com/annuh/reproducer-react-native

**Comment by Titozzz** [comment]
Hello this will be fixed in 0.73.1 indeed, it is the same bug as above

**Comment by cortinico** [comment]
Duplicates https://github.com/facebook/react-native/issues/41874
